src/main/java/org/reaktivity/nukleus/kafka/internal/stream/KafkaMergedFactory.java

```diff
@@ -819,6 +905,21 @@ private void onMergedInitialBegin(
 
             router.setThrottle(replyId, this::onMergedMessage);
 
+            final OctetsFW extension = begin.extension();
+            final DirectBuffer buffer = extension.buffer();
+            final int offset = extension.offset();
+            final int limit = extension.limit();
+
+            final KafkaBeginExFW beginEx = kafkaBeginExRO.wrap(buffer, offset, limit);
+            final KafkaMergedBeginExFW mergedBeginEx = beginEx.merged();
+            final Array32FW<KafkaFilterFW> filters = mergedBeginEx.filters();
+
+            this.maximumAge = filters.isEmpty() ||
+                filters.anyMatch(a -> !a.conditions().anyMatch(c -> c.kind() == KIND_AGE && c.age().get() == HISTORICAL)) ?
+                KafkaAge.LIVE : HISTORICAL;
```

Comment:
Suggest hoisting this logic to a separate method, reused in handling `FLUSH` as well.
```
this.maximumAge = asMaximumAge(filters);
```